core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -601,231 +457,330 @@ class SpendReportingServiceSpec
   }
 
-  it should "throw an exception when start date is after end date" in withDefaultTestDatabase {
-    dataSource: SlickDataSource =>
-      val service = createSpendReportingService(dataSource)
+  it should "throw an exception when user is not in alpha group" in {
+    val samDAO = mock[SamDAO]
+    when(samDAO.userHasAction(any(), any(), any(), any())).thenReturn(Future.successful(true))
+    when(
+      samDAO.userHasAction(
+        ArgumentMatchers.eq(SamResourceTypeNames.managedGroup),
+        ArgumentMatchers.eq("Alpha_Spend_Report_Users"),
+        ArgumentMatchers.eq(SamResourceAction("use")),
+        ArgumentMatchers.eq(testContext.userInfo)
+      )
+    ).thenReturn(Future.successful(false))
+
+    val service = new SpendReportingService(
+      testContext,
+      mock[SlickDataSource],
+      Resource.pure[IO, GoogleBigQueryService[IO]](mock[GoogleBigQueryService[IO]]),
+      samDAO,
+      spendReportingServiceConfig
+    )
 
-      val e = intercept[RawlsExceptionWithErrorReport] {
-        Await.result(service.getSpendForBillingProject(testData.billingProject.projectName,
-                                                       startDate = DateTime.now(),
-                                                       endDate = DateTime.now().minusDays(1)
-                     ),
-                     Duration.Inf
-        )
-      }
-      e.errorReport.statusCode shouldBe Option(StatusCodes.BadRequest)
+    val e = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        service.getSpendForBillingProject(billingProject.projectName, DateTime.now().minusDays(1), DateTime.now()),
+        Duration.Inf
+      )
+      fail("action was run without an exception being thrown")
+    }
+    e.errorReport.statusCode shouldBe Option(StatusCodes.Forbidden)
   }
 
-  it should s"throw an exception when date range is larger than ${spendReportingServiceConfig.maxDateRange} days" in withDefaultTestDatabase {
-    dataSource: SlickDataSource =>
-      val service = createSpendReportingService(dataSource)
-
-      val e = intercept[RawlsExceptionWithErrorReport] {
-        Await.result(
-          service.getSpendForBillingProject(testData.billingProject.projectName,
-                                            startDate =
-                                              DateTime.now().minusDays(spendReportingServiceConfig.maxDateRange + 1),
-                                            endDate = DateTime.now()
-          ),
-          Duration.Inf
-        )
-      }
-      e.errorReport.statusCode shouldBe Option(StatusCodes.BadRequest)
-  }
+  it should "throw an exception if the billing project cannot be found" in {
+    val samDAO = mock[SamDAO]
+    when(samDAO.userHasAction(any(), any(), any(), any())).thenReturn(Future.successful(true))
+    val dataSource = mock[SlickDataSource]
+    when(dataSource.inTransaction[Option[BillingProjectSpendExport]](any(), any())).thenReturn(Future.successful(None))
```

Comment:
Out of scope for this PR, but I'd like to start pushing more toward the repository pattern to avoid this slick datasource mocking (i.e., similar to what I've done [here](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingRepository.scala)).